src/Raven.Server/Documents/DatabaseRequestHandler.cs

```diff
@@ -60,51 +64,24 @@ public Task CheckForChanges(RequestHandlerContext context)
             return Task.CompletedTask;
         }
 
-        protected internal delegate void RefAction<T>(string databaseName, ref T configuration, JsonOperationContext context, ServerStore serverStore);
-
-        protected internal delegate Task<(long, object)> SetupFunc<T>(TransactionOperationContext context, string databaseName, T json, string raftRequestId);
-
-        protected Task DatabaseConfigurations(SetupFunc<BlittableJsonReaderObject> setupConfigurationFunc,
+        protected async Task DatabaseConfigurations(SetupFunc<BlittableJsonReaderObject> setupConfigurationFunc,
```

Comment:
about a year ago I made some changes to this code [here](
https://github.com/ravendb/ravendb/pull/12814/files#diff-ce0cb353d22ac1844f5a57c05f051e5e8cc73eba5561984bcf584b3f41bc1159)

in this PR I reverted those changes from last year - they are no longer needed since we added `AbstractHandlerProcessorForUpdateDatabaseConfiguration`

Reply:
Also, `PullReplicationHandler` is still using `DatabaseConfigurations` method, so we can't remove it yet :
https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Server/Documents/Handlers/PullReplicationHandler.cs#L30